src/v3/obfuscate.test.ts

```diff
@@ -0,0 +1,161 @@
+      const newDocument = await wrapDocument(testData, { version: SchemaId.v3 });
+      const { data, obfuscatedData } = obfuscateData(newDocument, fields);
+      const salts = decodeSalt(newDocument.proof.salts);
+      const salt1 = salts.find(s => s.path === fields[0]);
+      const salt2 = salts.find(s => s.path === fields[1]);
+      const value1 = get(newDocument, fields[0]);
+      const value2 = get(newDocument, fields[1]);
+      expect(obfuscatedData).toEqual([
+        toBuffer({ [fields[0]]: `${salt1?.value}:${value1}` }).toString("hex"),
+        toBuffer({ [fields[1]]: `${salt2?.value}:${value2}` }).toString("hex")
+      ]);
+      expect(decodeSalt(data.proof.salts)).not.toHaveProperty(fields);
+      expect(data).not.toHaveProperty(fields);
+    });
+
+    test("removes values from nested object", async () => {
+      const field = "template.name";
+      const newDocument = await wrapDocument(openAttestationData, { version: SchemaId.v3 });
+      const { data, obfuscatedData } = obfuscateData(newDocument, field);
+      const salt = decodeSalt(newDocument.proof.salts).find(s => s.path === field);
+      const value = get(newDocument, field);
+      expect(obfuscatedData).toEqual([toBuffer({ [field]: `${salt?.value}:${value}` }).toString("hex")]);
+      expect(decodeSalt(data.proof.salts)).not.toHaveProperty(field);
+      expect(data).not.toHaveProperty(field);
+    });
+
+    test("removes values from arrays", async () => {
+      const newDocument = await wrapDocument(openAttestationData, { version: SchemaId.v3 });
+      const fields = ["@context[2]", "@context[3]"];
+      const { data, obfuscatedData } = obfuscateData(newDocument, fields);
+      const salts = decodeSalt(newDocument.proof.salts);
+      const salt1 = salts.find(s => s.path === fields[0]);
+      const value1 = get(newDocument, fields[0]);
+      const salt2 = salts.find(s => s.path === fields[1]);
+      const value2 = get(newDocument, fields[1]);
+      expect(obfuscatedData).toEqual([
+        toBuffer({ [fields[0]]: `${salt1?.value}:${value1}` }).toString("hex"),
+        toBuffer({ [fields[1]]: `${salt2?.value}:${value2}` }).toString("hex")
+      ]);
+      expect(decodeSalt(data.proof.salts)).not.toHaveProperty(fields);
+      expect(data).not.toHaveProperty(fields);
+    });
+  });
+
+  describe("obfuscateDocument", () => {
+    test("is transitive", async () => {
+      const newDocument = await wrapDocument(testData, { version: SchemaId.v3 });
+      const intermediateDoc = obfuscateDocument(newDocument, "key1");
+      const finalDoc1 = obfuscateDocument(intermediateDoc, "key2");
+      const finalDoc2 = obfuscateDocument(newDocument, ["key1", "key2"]);
+      expect(finalDoc1).toEqual(finalDoc2);
+      expect(intermediateDoc).not.toHaveProperty("key1");
+      expect(finalDoc1).not.toHaveProperty(["key1", "key2"]);
+      expect(finalDoc2).not.toHaveProperty(["key1", "key2"]);
+    });
+
+    test("removes one field", async () => {
```

Comment:
this test the validity ? let's group that with the other similar test, same for other test

Reply:
Yep it tests the validity! So do I remove the ‘remove one field’, ‘removes multiple fields’, ‘removes values from nested object’ tests from ‘obfuscateDocument’ and leave the ‘is transitive’ one there? Since obfuscateDocument (above test suite) calls obfuscateData and the listed tests have been tested in there?

Reply:
oh, I didnt notice we were calling 2 different functions.

As is I dont think it's useful to call both obfuscateData and obfuscateDocument. I would call obfuscateDocument only and group both assertions made in both test (so one test calling obfuscateDocument, doing all the same assertions as in the 2 current tests)

Reply:
Make sure that all use cases are covered (i see there are more tests in obfuscateData than in obfuscateDocument)

Reply:
got it! not sure if i grouped them to how you would like, let me know after this PR:)